scripts/deploy_wallet_detector.js

```diff
@@ -2,33 +2,46 @@
 // Script to deploy a new ArgentWalletDectector.
 //
 // To deploy a new ArgentWalletDectector:
-// ./execute_script.sh deploy_wallet_detector.js <network>
+// bash ./scripts/execute_script.sh --no-compile scripts/deploy_wallet_detector.js <network>
 //
 // where:
 //    - network = [test, staging, prod]
 // ////////////////////////////////////////////////////////////////////
 
 /* global artifacts */
+global.web3 = web3;
 const ArgentWalletDetector = artifacts.require("ArgentWalletDetector");
 
-const { configurator, abiUploader } = require("deploy-manager.js");
+const deployManager = require("../utils/deploy-manager.js");
 
 const PROXYWALLET_CODEHASH = [
```

Comment:
I believe we have introduced a new `Proxy` when updating the `WalletFactory` to enable refund, and we are introducing a new one now with the `implementation` immutable. 

We shouldn't redeploy the `ArgentWalletDetector` but we need to update the existing instances and add these 2 `Proxy` bytecodes.

Reply:
This script was needed for `test` only which is why I kept it separate and not part of the 2.5 upgrade script which only calls `addImplementation`